src/OAuth1/AbstractProvider.php

```diff
@@ -135,12 +135,17 @@ public function parseToken($body)
      * @param array $parameters
      * @return \SocialConnect\Common\Http\Response
      */
-    protected function oauthRequest($uri, $method = Client::GET, $parameters = [])
+    public function oauthRequest($uri, $method = Client::GET, $parameters = [])
     {
         $headers = [
             'Accept' => 'application/json'
         ];
 
+        if (isset($parameters['headers']) && is_Array($parameters['headers'])) {
```

Comment:
Sorry,  didn't find time yesterday.

The idea is good, I just didn't want to interfere with the existing code that much.

And as f.e. Guzzle also provides multiple informations via a similar mechanism I thought it to be a perfect match. It's extensible without needing to alter code. 

In the ned it's your decision ;)